trinity/tx_pool/pool.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from typing import (
+    cast,
+    Iterable,
+    List
+)
+import uuid
+
+from bloom_filter import (
+    BloomFilter
+)
+
+from evm.rlp.transactions import (
+    BaseTransactionFields
+)
+from p2p.eth import (
+    Transactions
+)
+from p2p.peer import (
+    BasePeer,
+    ETHPeer,
+    PeerPool,
+    PeerPoolSubscriber,
+)
+from p2p.service import (
+    BaseService
+)
+
+
+class TxPool(BaseService, PeerPoolSubscriber):
+    """
+    The :class:`~trinity.tx_pool.pool.TxPool` class is responsible for holding and relaying
+    of transactions, represented as :class:`~evm.rlp.transactions.BaseTransaction` among the
+    connected peers.
+
+      .. note::
+
+        This is a minimal vialble implementation that only relays transactions but doesn't actually
+        hold on to them yet. It's still missing many features of a grown up transaction pool.
+    """
+    logger = logging.getLogger("trinity.tx_pool.TxPool")
+
+    def __init__(self, peer_pool: PeerPool) -> None:
+        super().__init__()
+        self._peer_pool = peer_pool
+        self._bloom = BloomFilter(max_elements=10000)
```

Comment:
some quick checks suggest `10,000` transactions is equivalent to about 90 blocks at current mainnet volume.

Lets bump this to something like 1 million, which looks like it will consume about 1mb of memory.  This also suggests a rolling bloom filter (maybe per peer) solution is going to be a priority very quickly.

Reply:
Will bump it up to 1m. I'd also like to check what geth and Parity do. If they do a rolling bloom filter or something else and how exactly. But yeah, can make that a high priority.